receivers/oracledb conflates IDs with Types

### Component(s)

`receiver/oracledb` 

### What happened?

## Description
Note that [`scraper.go`](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/oracledbreceiver/scraper.go#L77) passes in `id.String()` to `scraperhelper.NewScraper(...)`.  However, [`NewScraper`](https://github.com/open-telemetry/opentelemetry-collector/blob/412ed26684e2100d5367054a75d398d017c351b7/receiver/scraperhelper/scraper.go#L70) currently converts this string value to an ID of just the name `component.NewID(component.MustNewType(name)),`.

The net result of this is that the cannonical string for IDs with names (`type` + `/` + `name`) is treated as a (nameless, type-only) ID in `ScraperHelper`


This, combined with the new strict `Must`Blah methods, breaks discovery/observers, as the convention for observer/receiver_creator places `{end .  We [configure endpoints](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/receivercreator/runner.go#L71) as part of the name, and such are disallowed by the new conventions.

## Steps to Reproduce

## Expected Result

## Actual Result


### Collector version

v0.94

### Environment information

## Environment
OS: (e.g., "Ubuntu 20.04")
Compiler(if manually compiled): (e.g., "go 14.2")


### OpenTelemetry Collector configuration

_No response_

### Log output

```shell
From our downstream distro,


=== RUN   TestOracledbDockerObserver
<logs omitted due to length>
{"level":"debug","ts":1709071841.7361474,"caller":"receivercreator@v0.94.0/observerhandler.go:86","msg":"handling added endpoint","kind":"receiver","env":{"alternate_port":1521,"command":"","container_id":"45bee8d74a58b395274691f5890f05273cc527b5ef42d69b69cf164ea2afb18a","endpoint":"172.17.0.3:1521","host":"172.17.0.3","id":"45bee8d74a58b395274691f5890f05273cc527b5ef42d69b69cf164ea2afb18a:1521","image":"dca89199-34e9-49f9-a778-d43f641b9a1f","labels":{"io.buildah.version":"1.22.4-dev","org.opencontainers.image.source":"https://github.com/gvenzl/oci-oracle-xe","org.testcontainers":"true","org.testcontainers.lang":"go","org.testcontainers.sessionId":"de3f8a3f30676accfb1d6ed669da43f2714d64a2d7b8633de154fadfaa4f4a2a","org.testcontainers.version":"0.27.0"},"name":"oracledb","port":1521,"tag":"a7946821-71aa-406b-aa68-b7595114a48d","transport":"TCP","type":"container"}}
{"level":"debug","ts":1709071841.7362206,"msg":"evaluating statement","kind":"receiver","statement":{"Message":"starting receiver","Fields":{"caller":"receivercreator@v0.94.0/observerhandler.go:96","endpoint":"172.17.0.3:1521","endpoint_id":"45bee8d74a58b395274691f5890f05273cc527b5ef42d69b69cf164ea2afb18a:1521","kind":"receiver","name":"oracledb"},"Level":"info","Time":"2024-02-27T22:10:41.736203226Z","LoggerName":"","Caller":{"Defined":true,"PC":62074799,"File":"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/receivercreator@v0.94.0/observerhandler.go","Line":96,"Function":"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/receivercreator.(*observerHandler).OnAdd"},"Stack":""}}
{"level":"debug","ts":1709071841.7364082,"msg":"Adding watch_observer to embedded receiver config receiver attrs","kind":"receiver","observer":"docker_observer","receiver.type":"oracledb","receiver.name":"","receiverid":"oracledb","correp":"Endpoint{ID: 45bee8d74a58b395274691f5890f05273cc527b5ef42d69b69cf164ea2afb18a:1521, Target: 172.17.0.3:1521, Details: *observer.Container&{Name:oracledb Image:dca89199-34e9-49f9-a778-d43f641b9a1f Tag:a7946821-71aa-406b-aa68-b7595114a48d Port:1521 AlternatePort:1521 Command: ContainerID:45bee8d74a58b395274691f5890f05273cc527b5ef42d69b69cf164ea2afb18a Host:172.17.0.3 Transport:TCP Labels:map[io.buildah.version:1.22.4-dev org.opencontainers.image.source:https://github.com/gvenzl/oci-oracle-xe org.testcontainers:true org.testcontainers.lang:go org.testcontainers.sessionId:de3f8a3f30676accfb1d6ed669da43f2714d64a2d7b8633de154fadfaa4f4a2a org.testcontainers.version:0.27.0]}}"}
{"level":"debug","ts":1709071841.7364225,"msg":"non-strict matches will be evaluated with pattern map","kind":"receiver","map":"{\"endpoint\":\"172.17.0.3:1521\",\"endpoint_id\":\"45bee8d74a58b395274691f5890f05273cc527b5ef42d69b69cf164ea2afb18a:1521\",\"kind\":\"receiver\",\"message\":\"starting receiver\"}"}
panic: invalid character(s) in type "oracledb//receiver_creator/docker_observer{endpoint=\"172.17.0.3:1521\"}/45bee8d74a58b395274691f5890f05273cc527b5ef42d69b69cf164ea2afb18a:1521"

goroutine 212 [running]:
go.opentelemetry.io/collector/component.MustNewType(...)
	go.opentelemetry.io/collector/component@v0.94.1/config.go:149
go.opentelemetry.io/collector/receiver/scraperhelper.NewScraper({0xc000d4c480?, 0x68f61aa?}, 0xc002b2f710, {0xc0027b6b58, 0x2, 0xc000d4c360?})
	go.opentelemetry.io/collector/receiver@v0.94.1/scraperhelper/scraper.go:70 +0x13c
github.com/open-telemetry/opentelemetry-collector-contrib/receiver/oracledbreceiver.newScraper({{0x68f61aa?, 0x0?}, {0xc000d4c360?, 0x1000100010001?}}, 0xc002ad6c00, {{{0x0, 0x0}, {0x1, 0x0}, {0x0, ...}, ...}, ...}, ...)
	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/oracledbreceiver@v0.94.0/scraper.go:77 +0x328
github.com/open-telemetry/opentelemetry-collector-contrib/receiver/oracledbreceiver.createReceiverFunc.func1({0x7ffeed?, 0xc002aba640?}, {{{0x68f61aa, 0x8}, {0xc000d4c360, 0x83}}, {0xc002ac7380, {0x7585718, 0xc002ac1160}, {0x7585358, ...}, ...}, ...}, ...)
	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/oracledbreceiver@v0.94.0/factory.go:60 +0x2ef
go.opentelemetry.io/collector/receiver.CreateMetricsFunc.CreateMetricsReceiver(...)
	go.opentelemetry.io/collector/receiver@v0.94.1/receiver.go:129
github.com/open-telemetry/opentelemetry-collector-contrib/receiver/receivercreator.(*receiverRunner).createMetricsRuntimeReceiver(0xc0027b7460?, {0x75d2620, 0xc0006a4fa0}, {{0x68f61aa?, 0x45?}, {0xc000d4c360?, 0x75d2620?}}, {0x592a460, 0xc002b3e8c0}, {0x7579f30, ...})
	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/receivercreator@v0.94.0/runner.go:196 +0x273
github.com/open-telemetry/opentelemetry-collector-contrib/receiver/receivercreator.(*receiverRunner).start(0xc00043a700, {{{0xc0027bbd48, 0x8}, {0x0, 0x0}}, 0xc002b22d80, {0xc00058ee10, 0x45}}, 0xc00058ee10?, 0xc002b20b40)
	github.com/open-telemetry/opentelemetry-c
```
```


### Additional context

_No response_

--- a/.chloggen/oracledb_scraper_id.yaml
+++ b/.chloggen/oracledb_scraper_id.yaml
@@ -1,0 +1,27 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: bug_fix
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: oracledbreceiver
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Use metadata.Type for the scraper id to avoid invalid scraper IDs.
+
+# Mandatory: One or more tracking issues related to the change. You can use the PR number here if no issue exists.
+issues: [31457]
+
+# (Optional) One or more lines of additional information to render under the primary note.
+# These lines will be padded with 2 spaces and then inserted directly into the document.
+# Use pipe (|) for multiline entries.
+subtext:
+
+# If your change doesn't affect end users or the exported elements of any package,
+# you should instead start your pull request title with [chore] or use the "Skip Changelog" label.
+# Optional: The change log or logs in which this entry should be included.
+# e.g. '[user]' or '[user, api]'
+# Include 'user' if the change is relevant to end users.
+# Include 'api' if there is a change to a library API.
+# Default: '[user]'
+change_logs: []

--- a/receiver/oracledbreceiver/factory.go
+++ b/receiver/oracledbreceiver/factory.go
@@ -57,7 +57,7 @@ func createReceiverFunc(sqlOpenerFunc sqlOpenerFunc, clientProviderFunc clientPr
 			return nil, err
 		}
 
-		mp, err := newScraper(settings.ID, metricsBuilder, sqlCfg.MetricsBuilderConfig, sqlCfg.ScraperControllerSettings, settings.TelemetrySettings.Logger, func() (*sql.DB, error) {
+		mp, err := newScraper(metricsBuilder, sqlCfg.MetricsBuilderConfig, sqlCfg.ScraperControllerSettings, settings.TelemetrySettings.Logger, func() (*sql.DB, error) {
 			return sqlOpenerFunc(getDataSource(*sqlCfg))
 		}, clientProviderFunc, instanceName)
 		if err != nil {

--- a/receiver/oracledbreceiver/scraper.go
+++ b/receiver/oracledbreceiver/scraper.go
@@ -64,7 +64,7 @@ type scraper struct {
 	metricsBuilderConfig       metadata.MetricsBuilderConfig
 }
 
-func newScraper(id component.ID, metricsBuilder *metadata.MetricsBuilder, metricsBuilderConfig metadata.MetricsBuilderConfig, scrapeCfg scraperhelper.ScraperControllerSettings, logger *zap.Logger, providerFunc dbProviderFunc, clientProviderFunc clientProviderFunc, instanceName string) (scraperhelper.Scraper, error) {
+func newScraper(metricsBuilder *metadata.MetricsBuilder, metricsBuilderConfig metadata.MetricsBuilderConfig, scrapeCfg scraperhelper.ScraperControllerSettings, logger *zap.Logger, providerFunc dbProviderFunc, clientProviderFunc clientProviderFunc, instanceName string) (scraperhelper.Scraper, error) {
 	s := &scraper{
 		mb:                   metricsBuilder,
 		metricsBuilderConfig: metricsBuilderConfig,
@@ -74,7 +74,7 @@ func newScraper(id component.ID, metricsBuilder *metadata.MetricsBuilder, metric
 		clientProviderFunc:   clientProviderFunc,
 		instanceName:         instanceName,
 	}
-	return scraperhelper.NewScraper(id.String(), s.scrape, scraperhelper.WithShutdown(s.shutdown), scraperhelper.WithStart(s.start))
+	return scraperhelper.NewScraper(metadata.Type.String(), s.scrape, scraperhelper.WithShutdown(s.shutdown), scraperhelper.WithStart(s.start))
 }
 
 func (s *scraper) start(context.Context, component.Host) error {

